packages/indexer/src/starknet/parser.ts

```diff
@@ -0,0 +1,50 @@
+import { FieldElement } from "./felt";
+
+interface EventFilterParser {
+  eventFilter(name: string): Filter;
+  // eventParser(name: string): ParsedEvent[];
+}
+
+interface Filter {
+  name: string;
+}
+
+interface Event {
+  // Parsed event properties
+  event: any;
+  transaction: any;
+}
+
+export class Contract implements EventFilterParser {
+  readonly contractAddress: FieldElement;
+  readonly contractAbi: any[];
+
+  constructor(contractAddress: FieldElement, contractAbi: any[]) {
```

Comment:
Maybe we can import the `Abi` type from starknet.js?

That means we need to add it as dependency, but it's fine.

Reply:
Since this is the only place we need to use the `starknet.js` dependency, should we consider manually defining the Abi type in our codebase?

Reply:
I think adding it as dependency is fine. I expect to use it to decode events as well.